src/openrct2/peep/peep.c

```diff
@@ -9823,8 +9823,12 @@ static sint32 guest_path_find_leaving_park(rct_peep *peep, rct_map_element *map_
 	rct2_peep_spawn* peepSpawn = &gPeepSpawns[0];
 	// Peeps for whatever reason return to their original spawn point
 	// this in future should look for the nearest.
-	if (peep->sprite_index & 1 && gPeepSpawns[1].x != PEEP_SPAWN_UNDEFINED) {
-		peepSpawn++;
+	if (peep->sprite_index & 1) {
```

Comment:
This does not make sense anymore. You would need to rethink how this works. Probably just do a scenario Rand and clamp to max spawns and retry if invalid.